tests/testsuite/build.rs

```diff
@@ -137,6 +137,24 @@ fn incremental_config() {
             .with_status(0));
 }
 
+#[test]
+fn cargo_compile_with_workspace_excluded() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [package]
+            name = "foo"
+            version = "0.1.0"
+            authors = []
+        "#)
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    assert_that(
+        p.cargo("build").arg("--all").arg("--exclude").arg("foo"),
+        execs().with_stderr_does_not_contain("[..]virtual[..]")
```

Comment:
Let's write this with `with_stderr_contains`, so as to check the precise error message?

Reply:
will do!